src/components/ContactInput.tsx

```diff
@@ -52,75 +53,112 @@ const ContactInputComponent = ({
   const handleCountrySelect = (country: countryType) => {
     setFieldValue(`${prefix}.phone.country_code`, country.dialCode);
     setFilteredCountries([]);
+    setIsDropdownOpen(false);
+    setSearchTerm("");
   };
 
+  // Close dropdown when clicking outside
+  useEffect(() => {
+    const handleClickOutside = () => setIsDropdownOpen(false);
+    document.addEventListener("click", handleClickOutside);
+    return () => document.removeEventListener("click", handleClickOutside);
+  }, []);
+
+  // Get error messages
+  const countryCodeError = (errors as any)?.[prefix]?.phone?.country_code;
+  const phoneNumberError = (errors as any)?.[prefix]?.phone?.number;
+  const countryCodeTouched = (touched as any)?.[prefix]?.phone?.country_code;
+  const phoneNumberTouched = (touched as any)?.[prefix]?.phone?.number;
+
   return (
-    <div>
-      <p className="text-primary font-semibold">{label}</p>
+    <div className="mb-4">
+      <label className="text-primary font-semibold block mb-1">{label}</label>
       <div className="flex items-center w-full">
         {/* Country Code Input */}
-        <div className="relative w-20">
+        <div className="relative w-24">
           <Field
             component={FormikInputDiv}
             id={`${prefix}.phone.country_code`}
             name={`${prefix}.phone.country_code`}
             disabled={disabled}
             onChange={handleInputChange}
+            onClick={(e: React.MouseEvent) => {
+              e.stopPropagation();
+              setIsDropdownOpen(true);
+              if (!searchTerm) {
+                setFilteredCountries(sortedCountries);
+              }
+            }}
+            value={searchTerm}
             inputClass={`w-full ${
               zipClass ||
               "rounded-l-lg p-2 border border-r-0 border-primary bg-lightGray/30"
-            }`}
-            placeholder="code"
+            } ${countryCodeError && countryCodeTouched ? "border-red-500" : ""}`}
```

Comment:
unless I miss my mark this will cause 2 error messages to show up when the validation fails